sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -87,4 +92,93 @@ object StringUtils {
     }
     funcNames.toSeq
   }
+
+  /**
+   * Split the text into one or more SQLs with bracketed comments reserved
+   *
+   * Highlighted Corner Cases: semicolon in double quotes, single quotes or inline comments.
+   * Expected Behavior: The blanks will be trimed and a blank line will be omitted.
+   *
+   * @param text One or more SQLs separated by semicolons
+   * @return the trimmed SQL array (Array is for Java introp)
+   */
+  def split(text: String): Array[String] = {
+    val D_QUOTE: Char = '"'
+    val S_QUOTE: Char = '\''
+    val Q_QUOTE: Char = '`'
+    val SEMICOLON: Char = ';'
+    val ESCAPE: Char = '\\'
+    val DOT = '.'
+    val SINGLE_COMMENT = "--"
+    val BRACKETED_COMMENT_START = "/*"
+    val BRACKETED_COMMENT_END = "*/"
+    val FORWARD_SLASH = '/'
+
+    // quoteFlag acts as an enum of D_QUOTE, S_QUOTE, DOT
+    // * D_QUOTE: the cursor stands on a doulbe quoted string
+    // * S_QUOTE: the cursor stands on a single quoted string
+    // * DASH: the cursor stands in the SINGLE_COMMENT
+    // * FORWARD_SLASH: the cursor stands in the BRACKETED_COMMENT
+    // * DOT: default value for other cases
+    var quoteFlag: Char = DOT
+    var cursor: Int = 0
+    val ret: mutable.ArrayBuffer[String] = mutable.ArrayBuffer()
+    var currentSQL: mutable.StringBuilder = mutable.StringBuilder.newBuilder
+
+    while (cursor < text.length) {
+      val current: Char = text(cursor)
+
+      text.substring(cursor) match {
+        // if it stands on the opening of a bracketed comment, consume 2 characters
+        case remaining if quoteFlag == DOT
+                          && current == '/'
```

Comment:
Could you follow the indentation convention in our code base? https://github.com/databricks/scala-style-guide#indent

You can find many examples in the code base